src/client/ua_client_internal.h

```diff
@@ -40,6 +40,7 @@ typedef struct UA_Client_Subscription_s {
     LIST_HEAD(UA_ListOfClientMonitoredItems, UA_Client_MonitoredItem_s) MonitoredItems;
 } UA_Client_Subscription;
 
+void UA_Client_Subscriptions_forceDelete(UA_Client *client, UA_Client_Subscription *sub);
```

Comment:
Sorry, but no.
That will fail when subscriptions are disabled.

Why are you moving `UA_Client_Subscriptions_forceDelete` into the ifdef for subscrptions?
Why move the definition at all?